common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -311,6 +315,40 @@ static void copyMemory(Object src, long srcOffset, Object dst, long dstOffset, l
         }
     }
 
+    static boolean equals(final byte[] bytes1, final byte[] bytes2) {
+        if (bytes1.length != bytes2.length) {
+            return false;
+        }
+        if (bytes1.length == 0) {
+            return true;
+        }
+        if (!hasUnsafe() || !UNALIGNED) {
```

Comment:
Actually I think what we usually have in this class is we not check for hasUnsafe() ... This checks should be done in PlatformDependent which will only "delegate" to here if unsafe was found.